src/transport/PeerConnections.h

```diff
@@ -209,10 +209,12 @@ class PeerConnections
      * @return the state found, nullptr if not found
      */
     CHECK_RETURN_VALUE
-    PeerConnectionState * FindPeerConnectionState(Optional<NodeId> nodeId, uint16_t peerKeyId, PeerConnectionState * begin)
+    PeerConnectionState * FindPeerConnectionState(Optional<NodeId> nodeId, Optional<uint16_t> peerKeyId,
```

Comment:
This function is a bit confusing. Both lookup parameters are `Optional`. What happens if value of both the parameters is not provided? Do we need a more specific set of lookup functions?

Reply:
They are both indeed Optional values per spec.
FindPeerConnectionState has already taken care of cases when those values are not provided during lookup nodeId.ValueOr(kUndefinedNodeId), peerKeyId.ValueOr(0);

Reply:
By the way, `src/transport/PeerConnections.h` is already pretty broken.  See https://github.com/project-chip/connectedhomeip/issues/8481.

Regardless of what we do in this PR, we will need some significant rework there.

Reply:
ok, I can keep FindPeerConnectionState unchanged in this PR and leave to #8481 to address the broken issue.